app/src/main/java/cm/aptoide/pt/search/view/SearchPresenter.java

```diff
@@ -0,0 +1,301 @@
+package cm.aptoide.pt.search.view;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.util.Pair;
+import cm.aptoide.pt.R;
+import cm.aptoide.pt.crashreports.CrashReport;
+import cm.aptoide.pt.database.realm.MinimalAd;
+import cm.aptoide.pt.dataprovider.model.v7.DataList;
+import cm.aptoide.pt.dataprovider.model.v7.search.ListSearchApps;
+import cm.aptoide.pt.dataprovider.model.v7.search.SearchApp;
+import cm.aptoide.pt.presenter.Presenter;
+import cm.aptoide.pt.presenter.View;
+import cm.aptoide.pt.search.SearchAnalytics;
+import cm.aptoide.pt.search.SearchManager;
+import cm.aptoide.pt.search.SearchNavigator;
+import com.jakewharton.rxrelay.PublishRelay;
+import java.util.concurrent.TimeUnit;
+import rx.Observable;
+import rx.Scheduler;
+
+public class SearchPresenter implements Presenter {
+
+  private final SearchView view;
+  private final SearchAnalytics analytics;
+  private final SearchNavigator navigator;
+  private final CrashReport crashReport;
+  private final Scheduler viewScheduler;
+  private final SearchManager searchManager;
+  private final PublishRelay<MinimalAd> onAdClickRelay;
+  private final PublishRelay<SearchApp> onItemViewClickRelay;
+  private final PublishRelay<Pair<SearchApp, android.view.View>> onOpenPopupMenuClickRelay;
+
+  public SearchPresenter(SearchView view, SearchAnalytics analytics, SearchNavigator navigator,
+      CrashReport crashReport, Scheduler viewScheduler, SearchManager searchManager,
+      PublishRelay<MinimalAd> onAdClickRelay, PublishRelay<SearchApp> onItemViewClickRelay,
+      PublishRelay<Pair<SearchApp, android.view.View>> onOpenPopupMenuClickRelay) {
+    this.view = view;
+    this.analytics = analytics;
+    this.navigator = navigator;
+    this.crashReport = crashReport;
+    this.viewScheduler = viewScheduler;
+    this.searchManager = searchManager;
+    this.onAdClickRelay = onAdClickRelay;
+    this.onItemViewClickRelay = onItemViewClickRelay;
+    this.onOpenPopupMenuClickRelay = onOpenPopupMenuClickRelay;
+  }
+
+  @Override public void present() {
+    firstDataLoad();
+    loadAds();
```

Comment:
When results have no ad, and empty ad is added and show, with no actions.
Test: search for -> My Application

Reply:
fixed this issue